src/components/profiles/Profile.js

```diff
@@ -3,38 +3,30 @@ import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 import Loader from 'react-loader';
 import PropTypes from 'prop-types';
-import 'react-toastify/dist/ReactToastify.css';
 import profileImage from '../../assets/images/profileImage.png';
 import { getProfile } from '../../actions/userActions';
 
 export class Profile extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      loaded: false,
-    };
-  }
-
   componentDidMount() {
     const { getProfile } = this.props;
     getProfile();
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.getProfileInitiated === true) {
-      this.setState({ loaded: true });
+    if (nextProps.isLoggedIn === false) {
+      const { history } = this.props;
+      history.push('/login');
     }
   }
 
   render() {
-    const { loaded } = this.state;
-    const { getProfilePayload } = this.props;
-    const profileData = getProfilePayload;
+    const { loading } = this.props;
+    const { profilePayload } = this.props;
```

Comment:
Since `loading` and `profilePayload` are being picked from `props` you can pass them once same goes for `followers`, `following`, `bio` and `username`. Some thing like this:
```
    const {
      loading, profilePayload: {
        followers, following, username, bio,
      },
    } = this.props;
```

Reply:
@mendozabree , thanks for spotting that. The code has been refactored accordingly.